securedrop/journalist_app/admin.py

```diff
@@ -15,26 +18,30 @@
 from journalist_app.utils import (make_password, commit_account_changes, set_diceware_password,
                                   validate_hotp_secret, revoke_token)
 from journalist_app.forms import LogoForm, NewUserForm, SubmissionPreferencesForm
+from sdconfig import SDConfig
 
 
-def make_blueprint(config):
+def make_blueprint(config: SDConfig) -> Blueprint:
     view = Blueprint('admin', __name__)
 
     @view.route('/', methods=('GET', 'POST'))
     @admin_required
-    def index():
+    def index() -> str:
         users = Journalist.query.all()
         return render_template("admin.html", users=users)
 
     @view.route('/config', methods=('GET', 'POST'))
     @admin_required
-    def manage_config():
+    def manage_config() -> Union[str, werkzeug.Response]:
         # The UI prompt ("prevent") is the opposite of the setting ("allow"):
         submission_preferences_form = SubmissionPreferencesForm(
             prevent_document_uploads=not current_app.instance_config.allow_document_uploads)
         logo_form = LogoForm()
         if logo_form.validate_on_submit():
             f = logo_form.logo.data
+
+            if current_app.static_folder is None:
```

Comment:
Is it possible for `static_folder` to be `None`? It looks like we're always passing it in the `Flask` constructor in the source and journalist apps, and as a last resort, [`Flask` defaults it to "static"](https://flask.palletsprojects.com/en/1.1.x/api/#application-object).

Reply:
I'm not sure why but `current_app.static_folder` is typed as `Optional[str]`, so mypy will complain if None isn't checked here as the next line `join(current_app.static_folder etc.)` can't work if static_folder is None.

Reply:
This is the mypy error if you remove this:
```
securedrop\journalist_app\admin.py:43: error: Value of type variable "AnyStr" of "join" cannot be "Optional[str]"
```

Reply:
Ah, OK.